woocommerce/woocommerce-ios
# Store onboarding > launch store: UI integration

Part of #9122 


## Description


This PR integrated the new action `SiteAction.launchSite` to the UI `StoreOnboardingLaunchStoreView` through view model `StoreOnboardingLaunchStoreViewModel`. A new `@Published var` error was added to the view model, so that the error can be shown in an alert. As per the design decision in VyLr7LvKodHE4kINfBE7Lw-fi-1679%3A81721, we're only handling two error cases - when the site is already launched, or an unexpected error. The retry action for the unexpected error case will be handled separately as a nice-to-have. A callback `onLaunch` is triggered when the site is launched successfully, and a new success screen will be shown in a future PR. A log message is printed in the console in the success case for now.

## Testing instructions


Prerequisite: a site with an eCommerce or Business WPCOM plan that hasn't been launched yet

- Log in to the site in the prerequisite
- In the My store tab, tap on the `Launch your store` task --> the launch store screen should be presented
- Tap `Publish My Store` --> a spinner should be shown, and a message `🚀 site launched` should be shown in the Xcode console
- Dismiss the modal --> the `Launch your store` task should now be complete
- In the My store tab, tap on `View all (#)`, then tap `Launch your store` task again --> the launch store screen should be presented
- Tap `Publish My Store` --> a spinner should be shown, and an error alert should be shown that says the site is already published

## Screenshots


light | dark
-- | --
![Simulator Screen Shot - iPhone 14 - 2023-03-15 at 14 29 33](https://user-images.githubusercontent.com/1945542/225225447-a6aa692c-787d-4d12-a13a-48f7ef46106d.png) | ![Simulator Screen Shot - iPhone 14 - 2023-03-15 at 14 31 07](https://user-images.githubusercontent.com/1945542/225225452-d54ff9ee-6184-4830-8922-def9dc88cd91.png)


https://user-images.githubusercontent.com/1945542/225225215-7e2f87e5-33ae-456f-9694-3308dda86336.mov



---
